cool-demos/conditional-blocks.html

```diff
@@ -0,0 +1,452 @@
+
+            // "Thin" is a premium icon pack you get wth the cloud essential plan.
+            // See https://www.tiny.cloud for more details and purchasing options
+            // You can also create your own icon packs for TinyMCE
+            // https://www.tiny.cloud/docs/advanced/creating-an-icon-pack/
+            icons: 'thin',
+
+            // "Naked" is a premium icon pack you get wth the cloud essential plan
+            // See https://www.tiny.cloud for more details and purchasing options
+            // You can also create your own skins for TinyMCE
+            // https://www.tiny.cloud/docs/advanced/creating-a-skin/
+            skin: 'naked',
+
+            // If we don't need the menubar we can hide it
+            // https://www.tiny.cloud/docs/configure/editor-appearance/#menubar
+            menubar: false,
+
+            // Tell TinyMCE that condictional-block is a valid element.
+            // https://www.tiny.cloud/docs/configure/content-filtering/#custom_elements
+            custom_elements: "conditional-block",
+
+            setup: (editor) => {
+                // Instead of defining a custom icon pack, we can register individual
+                // icons using the API.
+                // https://www.tiny.cloud/docs/api/tinymce.editor.ui/tinymce.editor.ui.registry/#addicon
+                editor.ui.registry.addIcon('conditional-block', '<svg xmlns="http://www.w3.org/2000/svg" width="24" height="24"><path d="M19 4a2 2 0 1 1-1.854 2.751L15 6.75c-1.239 0-1.85.61-2.586 2.31l-.3.724c-.42 1.014-.795 1.738-1.246 2.217.406.43.751 1.06 1.12 1.92l.426 1.018c.704 1.626 1.294 2.256 2.428 2.307l.158.004h2.145a2 2 0 1 1 0 1.501L15 18.75l-.219-.004c-1.863-.072-2.821-1.086-3.742-3.208l-.49-1.17c-.513-1.163-.87-1.57-1.44-1.614L9 12.75l-2.146.001a2 2 0 1 1 0-1.501H9c.636 0 1.004-.383 1.548-1.619l.385-.92c.955-2.291 1.913-3.382 3.848-3.457L15 5.25h2.145A2 2 0 0 1 19 4z" fill-rule="evenodd"/></svg>');
+
+                // The init event is fired when the editor is fully loaded.
+                // https://www.tiny.cloud/docs/advanced/events/#editorcoreevents
+                editor.on("init", () => {
+                    // Get the iframe window object and the iframes document object
+                    // and call our setup function that creates the web component
+                    // https://www.tiny.cloud/docs/api/tinymce/tinymce.editor/#getwin
+                    // https://www.tiny.cloud/docs/api/tinymce/tinymce.editor/#getdoc
+                    const win = editor.getWin();
+                    const doc = editor.getDoc();
+                    setup(win, doc, editor);
+                });
+
+                // The preinit event is fired after the editor is loaded but before
+                // the content is loaded
+                // https://www.tiny.cloud/docs/advanced/events/#editorcoreevents
+                editor.on("preinit", () => {
```

Comment:
This is case insensitive so it doesn't really matter, but we normally keep custom event names using title case, eg: `PreInit`. That would also prevent confusion since it doesn't use a lowercase name.